src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -0,0 +1,186 @@
+        """
+        self.pool.connect(2)
+        csum = self.params.get("enable_checksum", '/run/container/*')
+        container = DaosContainer(self.context)
+        input_param = container.cont_input_values
+        input_param.enable_chksum = csum
+        container.create(poh=self.pool.pool.handle,
+                         con_prop=input_param)
+        container.open()
+        obj = DaosObj(self.context, container)
+        obj.create(objcls=1)
+        obj.open()
+        ioreq = IORequest(self.context,
+                          container,
+                          obj, objtype=4)
+        self.log.info("Writing the Single Dataset")
+        record_index = 0
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length[record_index])
+                d_key_value = "dkey {0}".format(dkey)
+                c_dkey = ctypes.create_string_buffer(d_key_value)
+                a_key_value = "akey {0}".format(akey)
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+                record_index = record_index + 1
+                if record_index == len(self.record_length):
+                    record_index = 0
+
+    def run_offline_reintegration_test(self, num_pool, data=False):
+        """Run the offline reintegration without data.
+           Args: num_pool (int) : Total pools to create
+                                  for testing purpose.
+                 data (bool) : False (Pool has no data) -Default
+                               True (Create some data in pool)
+           Returns : None
+        """
+        # Create a pool
+        pool = {}
+        pool_uuid = []
+        target_list = []
+        exclude_servers = len(self.hostlist_servers) - 1
+        nvme_size = 54000000000
+        scm_size = 6000000000
+
+        # Exclude target : random two targets
+        n = random.randint(1, 7)
+        target_list.append(n)
+        target_list.append(n+1)
+        t_string = "{},{}".format(target_list[0], target_list[1])
+
+        # Exclude rank : two ranks other than rank 0.
+        rank = random.randint(1, exclude_servers)
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(scm_size / num_pool)
+            pool[val].nvme_size.value = int(nvme_size / num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+            self.pool = pool[val]
+            if data is True:
```

Comment:
Just a suggestion, you could use this as a short hand instead:
`if data:`

Reply:
Done.